colcon_core/package_descriptor.py

```diff
@@ -51,14 +50,9 @@ def __init__(self, path):
         self.metadata = {}
 
     @property
-    def path(self):
-        """Get the path of the package."""
-        return self._path
-
-    @path.setter
-    def path(self, value):
-        self._path = Path(str(value))
-        self.realpath = os.path.realpath(str(value))
+    def realpath(self):
+        """Resolve the realpath of the package path."""
+        return os.path.realpath(str(self.path))
```

Comment:
If we don't want to cache the result why offer the property in the first place? Instead the property could be removed and in `__eq__` we call `os.path.realpath` directly.

Reply:
I didn't want to remove a public property, but now that I have your blessing....

Reply:
Usually a deprecation message would be better. In this case the property was only recently introduced and afaik there is no usage of it anywhere (beside the code removed in this PR).

Let's hope I don't regret taking this shortcut.